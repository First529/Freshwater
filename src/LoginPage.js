import React from "react";
import "./styles/login.css";

var googleImage = require("./images/google_logo.svg");
var facebookImage = require("./images/facebook-app-symbol.svg");

export default class LoginPage extends React.Component {
  render() {
    return (
      <div className="container">
        <div className="leftpane" />
        <div className="middlepane" />
        <div className="rightpane">
          <h1>Manatee</h1>
          <h3>A simple chat app</h3>
          <div className="send-name">
            {" "}
            <input placeholder="Enter your name" type="text" />{" "}
          </div>
          <p> or </p>
          <button className="container-google-color">
            <img src={googleImage} />
            <div className="container-google-text">
              {" "}
              Sign in with Google account{" "}
            </div>
          </button>
          <button className="container-facebook-color">
            <img src={facebookImage} />
            <div className="container-facebook-text">
              {" "}
              Sign in with Facebook account{" "}
            </div>
          </button>
        </div>
      </div>
    );
  }
}
