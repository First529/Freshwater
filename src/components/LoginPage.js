import React from "react";
import "../styles/login.css";

import google_logo from "../images/google_logo.svg"
import facebook_logo from "../images/facebook_logo.svg"

export default class LoginPage extends React.Component {
  render() {
    return (
      <div className="container">
        <div className="leftpane" />
        <div className="middlepane" />
        <div className="rightpane">
          <h1>Manatee</h1>
          <h3>A simple chat app</h3>
          <div className="send-name">
            {" "}
            <input placeholder="Enter your name" type="text" />{" "}
          </div>
          <p> or </p>
          <button className="container-google-color">
            <img src={google_logo} alt={"google_logo"}/>
            <div className="container-google-text">
              {" "}
              Sign in with Google account{" "}
            </div>
          </button>
          <div className="empty-space" />
          <button className="container-facebook-color">
            <img src={facebook_logo} alt={"facebook_logo"}/>
            <div className="container-facebook-text">
              {" "}
              Sign in with Facebook account{" "}
            </div>
          </button>
        </div>
      </div>
    );
  }
}
