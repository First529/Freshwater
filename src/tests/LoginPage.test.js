import React from "react";
import { shallow } from "enzyme";
import LoginPage from "../components/LoginPage";

describe("LoginPage", () => {
  it("renders correctly", () => {
    const component = shallow(<LoginPage />);
    expect(component).toMatchSnapshot();
  });

  it("check if username is empty", () => {
    const component = shallow(<LoginPage />);
    expect(component.state("username")).toBe("");
  });
});
