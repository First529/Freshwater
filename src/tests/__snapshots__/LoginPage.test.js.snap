// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="leftpane"
  />
  <div
    className="middlepane"
  />
  <div
    className="rightpane"
  >
    <h1>
      Manatee
    </h1>
    <h3>
      A simple chat app
    </h3>
    <div
      className="send-name"
    >
       
      <input
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Enter your name"
        type="text"
        value=""
      />
       
    </div>
    <div
      id="extra"
    >
      <p>
         or 
      </p>
      <button
        className="container-google-color"
      >
        <img
          alt="google_logo"
          src="google_logo.svg"
        />
        <div
          className="container-google-text"
        >
           
          Sign in with Google account
           
        </div>
      </button>
      <div
        className="empty-space"
      />
      <button
        className="container-facebook-color"
      >
        <img
          alt="facebook_logo"
          src="facebook_logo.svg"
        />
        <div
          className="container-facebook-text"
        >
           
          Sign in with Facebook account
           
        </div>
      </button>
    </div>
    <BrowserRouter>
      <Link
        id="submit"
        replace={false}
        to={
          Object {
            "pathname": "/public-room",
            "state": Object {
              "fromUserName": "",
            },
          }
        }
      />
    </BrowserRouter>
  </div>
</div>
`;
